core/runtime/src/main/java/io/quarkiverse/flow/providers/CredentialsProviderSecretManager.java

```diff
@@ -0,0 +1,153 @@
+ * <li>else fail with guidance</li>
+ * </ol>
+ */
+@ApplicationScoped
+public class CredentialsProviderSecretManager implements SecretManager {
+
+    private static final String ROOT = FlowSecretsConfig.ROOT_KEY;
+    private static final String GLOBAL_KEY = ROOT + ".credentials-provider-name";
+    private static final String PER_SECRET_KEY = ROOT + ".credentials-provider-names.<secret>";
+    private final Map<String, CredentialsProvider> providerBeanCache = new ConcurrentHashMap<>();
+    private final Map<String, CredentialsProvider> secretCache = new ConcurrentHashMap<>();
+    @Inject
+    FlowSecretsConfig flowSecrets;
+    @Inject
+    @Any
+    Instance<CredentialsProvider> providers;
+    private volatile boolean hasAnyProviders;
+    private volatile Optional<CredentialsProvider> singleProvider = Optional.empty();
+
+    private static String nullSafeTrim(String v) {
+        if (v == null)
+            return null;
+        String t = v.trim();
+        return t.isEmpty() ? null : t;
+    }
+
+    @PostConstruct
+    void init() {
+        final List<CredentialsProvider> all = providers.stream().toList();
+        hasAnyProviders = !all.isEmpty();
+
+        if (providers.isResolvable() && !providers.isAmbiguous()) {
+            singleProvider = Optional.of(providers.get());
+        }
+
+        for (CredentialsProvider p : all) {
+            Named n = p.getClass().getAnnotation(Named.class);
+            if (n != null && !n.value().isBlank()) {
+                providerBeanCache.putIfAbsent(n.value(), p);
+            }
+        }
+    }
+
+    /**
+     * Recorder checks this to decide whether to wire this SM or let the SDK fallback handle resolution.
+     */
+    public boolean hasAnyProviders() {
+        return hasAnyProviders;
+    }
+
+    @Override
+    public Map<String, String> secret(String secretName) {
+        CredentialsProvider provider = secretCache.computeIfAbsent(secretName, this::resolveProviderForSecret);
+        Map<String, String> creds = provider.getCredentials(secretName);
+        if (creds == null || creds.isEmpty()) {
+            throw new IllegalStateException("No credentials found for secret '" + secretName
+                    + "' from provider @" + namedOf(provider) + " (" + provider.getClass().getName() + ")");
+        }
+        return creds;
+    }
+
+    private CredentialsProvider resolveProviderForSecret(String secretName) {
+        String per = nullSafeTrim(flowSecrets.credentialsProviderNames().get(secretName));
+        if (per != null) {
+            return selectConfigProvider(per);
+        }
+
+        Optional<String> global = flowSecrets.credentialsProviderName().map(String::trim).filter(s -> !s.isEmpty());
+        if (global.isPresent()) {
+            return selectConfigProvider(global.get());
+        }
+
+        if (singleProvider.isPresent()) {
+            return singleProvider.get();
+        }
+
+        throw new IllegalStateException(
```

Comment:
In this case I have doubts, contrary to the previous case, this indicates a configuration problem that requires user intervention, so throwing the exception is the way to go